ArduSub/control_motordetect.cpp

```diff
@@ -0,0 +1,175 @@
+#include "Sub.h"
+#include "stdio.h"
+
+/*
+ * control_motordetect.cpp - init and run calls for motordetect flightmode;
+ *
+ *  This mode pulses all thrusters to detect if they need to be reversed.
+ *  This still requires that the user has the correct frame selected and the motors
+ *  are connected to the correct ESCs.
+ *
+ *  For each motor:
+ *      wait until vehicle is stopped for > 500ms
+ *      apply throttle up for 500ms
+ *      If results are good:
+ *          save direction and try the next motor.
+ *      else
+ *          wait until vehicle is stopped for > 500ms
+ *          apply throttle down for 500ms
+ *          If results are good
+ *              save direction and try the next motor.
+ *          If results are bad
+ *              Abort!
+ */
+
+// controller states
+enum test_state {
```

Comment:
wrap these globals in an anonymous namespace, then they will be visible to only this file.

Reply:
done